src/components/App.js

```diff
@@ -1,14 +1,42 @@
+/* eslint-disable react/destructuring-assignment */
 import React from 'react';
 import ButtonPanel from './ButtonPanel';
 import Display from './Display';
+import calculate from '../logic/calculate';
 
-const App = () => {
-  return (
-    <div className="CalculatorApp">
-      <Display />
-      <ButtonPanel />
-    </div>
-  );
-};
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      next: null,
+      operation: null,
+      total: null,
+    };
+  }
+
+  handleClick(buttonName) {
+    const { next, operation, total } = calculate(this.state, buttonName);
+    this.setState({
+      next,
+      operation,
+      total,
+    });
+  }
+
+  render() {
+    let value = null;
+    if (this.state.operation === null) {
+      value = this.state.total ? this.state.total : '0';
+    } else {
+      value = this.state.next ? this.state.next : this.state.total;
+    }
+    return (
+      <div className="CalculatorApp">
+        <Display value={value} />
```

Comment:
[Optional]

We could also conditionally pass `next` or `total` as `Display` `value` props.

```jsx
  render() {
    const { next, total } = this.state;
    return (
      <div className="CalculatorApp">
        <Display value={next || total} />
        <ButtonPanel onClick={(buttonName) => this.handleClick(buttonName)} />
      </div>
    );
  }
```

Note that doing this should also fix the rule that was disabled above: `/* eslint-disable react/destructuring-assignment */`